src/main/java/uk/gov/justice/digital/delius/transformers/AssessmentTransformer.java

```diff
@@ -6,39 +6,46 @@
 import uk.gov.justice.digital.delius.jpa.standard.entity.Offender;
 
 import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
 import java.util.Optional;
+import java.util.stream.Stream;
 
 public class AssessmentTransformer {
 
-    public static OffenderAssessments assessmentsOf(Offender offender, Optional<OGRSAssessment> OGRSAssessment, Optional<OASYSAssessment> OASYSAssessment) {
+    public static OffenderAssessments assessmentsOf(Offender offender, Optional<OGRSAssessment> ogrsAssessment, Optional<OASYSAssessment> oasysAssessment) {
         return OffenderAssessments
             .builder()
             .rsrScore(offender.getDynamicRsrScore())
-            .ogrsScore(getOGRSScore(OGRSAssessment, OASYSAssessment))
+            .ogrsScore(getOGRSScore(ogrsAssessment, oasysAssessment))
+            .orgsLastUpdate(getLastUpdateDate(ogrsAssessment,oasysAssessment ))
             .build();
     }
-
-    private static Integer getOGRSScore(Optional<OGRSAssessment> OGRSAssessment, Optional<OASYSAssessment> OASYSAssessment) {
-
-        Integer OASYSAssessmentScore = OASYSAssessment.map(o -> o.getOGRSScore2()).orElse(null);
-        LocalDate OASYSAssessmentDate = OASYSAssessment.map(o -> o.getAssessmentDate()).orElse(null);;
-
-        return OGRSAssessment.map(OGRS -> {
-            final var OGRSAssessmentScore = OGRS.getOGRS3Score2();
-            if (null == OASYSAssessmentScore) {
-                return OGRSAssessmentScore;
-            }
-
-            if (null != OGRSAssessmentScore) {
-                final var OGRSAssessmentDate = OGRS.getAssessmentDate();
-                if (OGRSAssessmentDate.isAfter(OASYSAssessmentDate)) {
-                    return OGRSAssessmentScore;
+    private static LocalDate getLastUpdateDate( Optional<OGRSAssessment> ogrsAssessment, Optional<OASYSAssessment> oasysAssessment){
+        var val1 = ogrsAssessment.map(OGRSAssessment::getLastUpdatedDate).orElse(null);
+        var val2 = oasysAssessment.map(OASYSAssessment::getLastUpdatedDate).orElse(null);
+        if (val1 != null && val2 != null) {
+            return Stream.of(val1, val2).max(LocalDate::compareTo).orElse(null);
+        }
+        return Optional.ofNullable(val1).orElse(val2);
```

Comment:
Looks good, you could simplify this though - no need for null-checks as `flatMap(Optional::stream)` can do that for you e.g.

```suggestion
        var ogrsDate = ogrsAssessment.map(OGRSAssessment::getLastUpdatedDate);
        var oasysDate = oasysAssessment.map(OASYSAssessment::getLastUpdatedDate);
        return Stream.of(ogrsDate, oasysDate).flatMap(Optional::stream).max(LocalDate::compareTo).orElse(null);
```